### PR TITLE
Evaluate using System.Text.Json instead of NewtonSoft.Json

### PR DESCRIPTION
**Pull request by theczorn**
* Patched JsonVersionService to work
* replaced newtonsoft.json with dotnet System.Text.Json
* Removed unusued using statements across code

### CODE REVIEW

File: src/BlackSlope.Api.Common/Versioning/Services/JsonVersionService.cs [hunk 1]
```diff
@@ -18,10 +18,9 @@ public JsonVersionService(IFileSystem fileSystem, IWebHostEnvironment hostingEnv
 
         public Version GetVersion()
         {
-            var filepath = _fileSystem.Path.Combine(_hostingEnvironment.ContentRootPath, "Common", "Version", "version.json");
+            var filepath = _fileSystem.Path.Combine(_hostingEnvironment.ContentRootPath, "..", "Blackslope.Api.Common", "Versioning", "version.json");
```
Comment (theczorn) [inline src/BlackSlope.Api.Common/Versioning/Services/JsonVersionService.cs:21]: File structure was off causing failures.

File: src/BlackSlope.Api.Common/Versioning/VersionJsonConverter.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace BlackSlope.Api.Common.Versioning
+{
+    public class VersionJsonConverter : JsonConverter<Version>
```
Comment (theczorn) [inline src/BlackSlope.Api.Common/Versioning/VersionJsonConverter.cs:7]: One of the limitations of System.Text.Json is in Converter logic.

Normally I'd leave all this alone, but the `Version` object structure was intended to be immutable. I tried to adhere to this as much as possible.

See [this conversion guide](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#deserialize-to-immutable-classes-and-structs)

**Comment by theczorn** [comment]
https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to

Calls out a lot of gotchas. From what I've seen I think this will work well in 95% of scenarios. Some minor quirks are:
a.) Not throwing on required fields with `NULL` values: [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#required-properties)
b.) Limitations on manipulating JSON DOM by shunting you into the JsonDocument type: [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#how-to-search-a-jsondocument-and-jsonelement-for-sub-elements)

I think generally the new libraries will work just fine. If you're doing any manual manipulation of JSON data and want to be lazy/"dynamic" about it from what I gather you are severely limited vs Newtonsoft. The design goal aims towards strongly typed serialization/deserialization. So often times the solution to your consumption woes is to just write another `Converter()` subclass to handle your edgecases.

As an aside, I swore I found some documentation allowing you to run both Newtonsoft and Text.Json side-by-side. I'll post it here for discussion when I find it again.

File: src/BlackSlope.Api.Common/Versioning/Services/JsonVersionService.cs (hunk 1 above)
Comment (pmotameni) [inline src/BlackSlope.Api.Common/Versioning/Services/JsonVersionService.cs:21]: Please create a new issue to fix this. The plan is to make 'Common' a reusable nuget package, this approach does not work.

Comment (theczorn) [reply src/BlackSlope.Api.Common/Versioning/Services/JsonVersionService.cs:21]: Done.

https://github.com/SlalomBuild/blackslope.net/issues/35

File: src/BlackSlope.Api.Common/Versioning/VersionJsonConverter.cs (hunk 2 above)
Comment (andykongchi) [reply src/BlackSlope.Api.Common/Versioning/VersionJsonConverter.cs:7]: Can we get some unit tests added for this VersionJsonConverter.cs logic. Thanks

File: src/BlackSlope.Api/Startup.cs [hunk 3]
```diff
@@ -39,7 +39,11 @@ public void ConfigureServices(IServiceCollection services)
             services.AddAutoMapper(GetAssemblyNamesToScanForMapperProfiles());
             services.AddCorrelation();
             services.AddTransient<IFileSystem, FileSystem>();
-            services.AddTransient<IVersionService, AssemblyVersionService>();
+
+            // TODO: Pick one of the below versioning services
```
Comment (andykongchi) [inline src/BlackSlope.Api/Startup.cs:43]: Is this todo intended for this PR or a separate issue?

**Review by andykongchi — CHANGES_REQUESTED** [review]

File: src/BlackSlope.Api/Startup.cs (hunk 3 above)
Comment (theczorn) [reply src/BlackSlope.Api/Startup.cs:43]: I'm going to switch this over to `NOTE:`. It's not really an item to fix as much as a "Hey, configure this for a new proj" thing.

Comment (theczorn) [reply src/BlackSlope.Api/Startup.cs]: Done.

**Comment by theczorn** [comment]
Tests have been added. @andykongchi

**Review by andykongchi — APPROVED** [review]

**Pull request merged**